src/twisted/application/runner/test/test_pidfile.py

```diff
@@ -75,182 +73,188 @@ class PIDFileTests(twisted.trial.unittest.TestCase):
         """
         pid = 1995
 
-        pidFile = PIDFile(DummyFilePath())
+        pidFile = PIDFile(self.filePath())
         pidFile._write(pid)
 
         self.assertEqual(pidFile.read(), pid)
 
-    def test_writePIDInvalid(self):
+    def test_writePIDInvalid(self) -> None:
         """
         L{PIDFile._write} raises L{ValueError} when given an invalid PID.
         """
-        pidFile = PIDFile(DummyFilePath())
+        pidFile = PIDFile(self.filePath())
 
         self.assertRaises(ValueError, pidFile._write, "burp")
 
-    def test_writeRunningPID(self):
+    def test_writeRunningPID(self) -> None:
         """
         L{PIDFile.writeRunningPID} stores the PID for the current process.
         """
-        pidFile = PIDFile(DummyFilePath())
+        pidFile = PIDFile(self.filePath())
         pidFile.writeRunningPID()
 
         self.assertEqual(pidFile.read(), getpid())
 
-    def test_remove(self):
+    def test_remove(self) -> None:
         """
         L{PIDFile.remove} removes the PID file.
         """
-        pidFile = PIDFile(DummyFilePath(b""))
+        pidFile = PIDFile(self.filePath(b""))
         self.assertTrue(pidFile.filePath.exists())
 
         pidFile.remove()
         self.assertFalse(pidFile.filePath.exists())
 
     @ifPlatformSupported
-    def test_isRunningDoesExist(self):
+    def test_isRunningDoesExist(self) -> None:
         """
         L{PIDFile.isRunning} returns true for a process that does exist.
         """
-        pidFile = PIDFile(DummyFilePath())
+        pidFile = PIDFile(self.filePath())
         pidFile._write(1337)
 
-        def kill(pid, signal):
+        def kill(pid: int, signal: int) -> None:
             return  # Don't actually kill anything
 
         self.patch(_pidfile, "kill", kill)
 
         self.assertTrue(pidFile.isRunning())
 
     @ifPlatformSupported
-    def test_isRunningThis(self):
+    def test_isRunningThis(self) -> None:
         """
         L{PIDFile.isRunning} returns true for this process (which is running).
 
         @note: This differs from L{PIDFileTests.test_isRunningDoesExist} in
         that it actually invokes the C{kill} system call, which is useful for
         testing of our chosen method for probing the existence of a process.
         """
-        pidFile = PIDFile(DummyFilePath())
+        pidFile = PIDFile(self.filePath())
         pidFile.writeRunningPID()
 
         self.assertTrue(pidFile.isRunning())
 
     @ifPlatformSupported
-    def test_isRunningDoesNotExist(self):
+    def test_isRunningDoesNotExist(self) -> None:
         """
         L{PIDFile.isRunning} raises L{StalePIDFileError} for a process that
         does not exist (errno=ESRCH).
         """
-        pidFile = PIDFile(DummyFilePath())
+        pidFile = PIDFile(self.filePath())
         pidFile._write(1337)
 
-        def kill(pid, signal):
+        def kill(pid: int, signal: int) -> None:
             raise OSError(errno.ESRCH, "No such process")
```

Comment:
Maybe use `NoReturn` instead?

Reply:
The thing it is emulating returns, though

Reply:
As far as I know, `NoReturn` is compatible with any return type, so it shouldn't be an issue for mypy. But perhaps sticking to the emulated signature is indeed clearer.